src/averageAges.js

```diff
@@ -54,11 +65,20 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
+
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const children = people.map((person) => (
+    {
+      ...person,
+      motherObject: people
+        .find((motherName) => motherName.name === person.mother),
+    }))
+    .filter(person => person.motherObject && (onlyWithSon === undefined
+  || person.sex === 'm'));
```

Comment:
It's difficult to read, motherObject looks strange, it can be just mother. Also, I think you actually don't need this transformation, you can just filter people that have mothers.

Reply:
about "It's difficult to read, motherObject looks strange, it can be just mother"  - I totally agree.
about "Also, I think you actually don't need this transformation, you can just filter people that have mothers." - in line 80 I get mother's "born", so it would be the second time I look for mother. I think searching for the mother just once is better, so I'm saving the first search results.